docs/index.bs

```diff
@@ -3237,30 +3239,78 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
+      1. If |condition|["{{RouterCondition/requestDestination}}"] [=map/exists=], set |hasCondition| to true.
+      1. If |condition|["{{RouterCondition/runningStatus}}"] [=map/exists=], set |hasCondition| to true.
+      1. If |condition|["{{RouterCondition/_or}}"] [=map/exists=], then:
+          1. If |hasCondition| is true, return false.
+
+              Note: For ease of understanding the router rule, the "or" condition is mutual exclusive with other conditions.
+
+          1. Let |or conditions| be |condition|["{{RouterCondition/_or}}"].
+          1. For each |or condition| of |or conditions|:
+              1. If running [=Verify Router Condition=] algorithm with |or condition| and |serviceWorker| returns false, return false.
+          1. Set |hasCondition| to true.
       1. Return |hasCondition|.
   </section>
 
+  <section algorithm>
+    <h3 id="match-router-condition-algorithm"><dfn>Match Router Condition</dfn></h3>
+      : Input
+      :: |condition|, a {{RouterCondition}}
+      :: |serviceWorker|, a [=/service worker=]
+      :: |request|, a [=/request=]
+      : Output
+      :: a boolean
+
+      Note: if there are multiple conditions, all conditions will be matched to return the source.
+
+      1. If |condition|["{{RouterCondition/urlPattern}}"] [=map/exists=], then:
+          1. Let |rawPattern| be |condition|["{{RouterCondition/urlPattern}}"].
+          1. Let |pattern| be the result of running <a>Parse URL Pattern</a> algorithm passing |rawPattern| and |serviceWorker|.
+          1. If running [=match=] with |pattern| and |request|'s [=request/URL=] returns null, return false.
+      1. If |condition|["{{RouterCondition/requestMethod}}"] [=map/exists=], then:
+          1. Let |method| be |condition|["{{RouterCondition/requestMethod}}"].
+          1. If |request|'s [=request/method=] is not |method|, return false.
+      1. If |condition|["{{RouterCondition/requestMode}}"] [=map/exists=], then:
+          1. Let |mode| be |condition|["{{RouterCondition/requestMode}}"].
+          1. If |request|'s [=request/mode=] is not |mode|, return false.
+      1. If |condition|["{{RouterCondition/requestDestination}}"] [=map/exists=], then:
+          1. Let |destination| be |condition|["{{RouterCondition/requestDestination}}"].
+          1. If |request|'s [=request/destination=] is not |destination|, return false.
+      1. If |condition|["{{RouterCondition/runningStatus}}"] [=map/exists=], then:
+          1. Let |runningStatus| be |condition|["{{RouterCondition/runningStatus}}"].
+          1. If |runningStatus| is {{RunningStatus/"running"}}, and |serviceWorker| is not [=running=], return false.
+          1. If |runningStatus| is {{RunningStatus/"not-running"}}, and |serviceWorker| is [=running=], return false.
+      1. If |condition|["{{RouterCondition/_or}}"] [=map/exists=], then:
+          1. Let |or conditions| be |condition|["{{RouterCondition/_or}}"].
+          1. Let |or condition matched| be false.
```

Comment:
Nit: you can simplify this as follows:

> 1. For each orCondition of orConditions:
>    1. If running ... returns true, then return true.
> 1. Return false.

Reply:
Ah, it reminds me https://github.com/WICG/service-worker-static-routing-api/issues/9
`or: []` always returns false.
I have applied the change.